pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -328,6 +292,39 @@ public TimeZone get() {
     replicationConnection = PGProperty.REPLICATION.get(info) != null;
   }
 
+  private Set<Integer> getBinaryOids(Properties info) throws PSQLException {
+    boolean binaryTransfer = PGProperty.BINARY_TRANSFER.getBoolean(info);
+    // Formats that currently have binary protocol support
+    Set<Integer> binaryOids = new HashSet<Integer>();
```

Comment:
consider initializing this to a reasonable size to limit (or eliminate) the need to reallocate the backing array. the default starting size of 16 currently guarantees (at least) 1 reallocation each time this method is called.

Reply:
made changes. Set initial capacity to 32 (20 values if `binaryTransfer` enabled + some values from BINARY_TRANSFER_ENABLE)